### PR TITLE
Animals joins

### PR DESCRIPTION
**Pull request by ismailmunyentwari9**
# What I accomplished in this part of project 🥇 

- [ ] Created new PR as requested by reviewer comparing changes between `dev` and `animals_joins` branches <a href ="https://github.com/ismailmunyentwari9/Vet-clinic-database/pull/4"> and this is invalid PR </a> which was comparing changes between `main` and a`nimals_joins` 
- [ ] Create a table named `vets`
- [ ] Create a "join table" called `specializations`
- [ ] Create a "join table" called `visits`
- [ ] Write queries to answer the following:
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(91).png"> Who was the last animal seen by William Tatcher?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(92).png">How many different animals did Stephanie Mendez see?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(93).png">List all vets and their specialties, including vets with no specialties.</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(94).png">List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(95).png">What animal has the most visits to vets?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(103).png">Who was Maisy Smith's first visit?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(98).png">Details for most recent visit: animal information, vet information, and date of visit.</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(99).png">How many visits were with a vet that did not specialize in that animal's species?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_joins/Screenshot%20(104).png">What specialty should Maisy Smith consider getting? Look for the species she gets the most.</a>

# 🧠 Skills Gained 🍾 

- [ ] Understand the different types of relationships between tables.
- [ ] Prepare complex queries that answer analytical questions.
- [ ] Use primary key & foreign key mechanism for joining tables.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,3 +1,227 @@
...
+JOIN animals ON visits.animal_id = animals.id
+GROUP BY animals.id
+ORDER BY num_visits DESC
+LIMIT 1;
+
+--Who was Maisy Smith's first visit?
+SELECT animals.name, visits.visit_date from visits
+JOIN animals ON visits.animal_id = animals.id
+ORDER BY visit_date ASC
+LIMIT 1;
```
Comment (addod19) [inline queries.sql]: - [ ] This part of the query result is not as expected as we can see in the project requirements

Screenshot from 2023-05-04 15-35-19

File: queries.sql [hunk 2]
```diff
@@ -1,3 +1,227 @@
...
+SELECT s.name AS specialty, COUNT(*) AS visits_count
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+JOIN species s ON a.species_id = s.id
+JOIN specialization sp ON sp.species_id = s.id
+JOIN vets vt ON v.vet_id = vt.id
+WHERE a.name = 'Maisy Smith'
+GROUP BY s.name
+ORDER BY visits_count DESC
+LIMIT 1;
```
Comment (addod19) [inline queries.sql]: - [ ] This pat of the query is not giving out the right result, let us kindly check and fix it
Screenshot from 2023-05-04 15-43-04

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @ismailmunyentwari9, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Nice job so far with the project :rocket: 


 ## Required Changes ♻️
_Check the comments under the review._
- [ ] Kindly use only the screenshot used for a specific query, it's quite confusing looking at the screenshots

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (ismailmunyentwari9) [reply queries.sql]: Hello Sir, may you please give me hint on that last change you requested

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @ismailmunyentwari9 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**